### DISCUSSION TITLE
ERROR: TypeError: title.trim is not a function

### DISCUSSION THREAD

**Discussion by bryanwhiting** [quarto-dev/quarto-cli#8584] [Q&A] [answered]
### Description

Posting for posterity. I was getting this error, which wasn't helpful.
```
❯ quarto render index.qmd 
ERROR: TypeError: title.trim is not a function
```

But the issue was that I had a ```title: 2000``` (a number). When i swtich to ```title: "2000"``` everything processed.

**Comment by bryanwhiting** [comment, marked as answer]
But the issue was that I had a title: 2000 (a number). When i swtich to title: "2000" everything processed.

**Comment by cscheid** [comment]
That's strange - our validation should have caught this.

**Reply by cscheid** [reply]
I can't reproduce your issue:

````
$ cat 8584.qmd
---
title: 2000
format: html
---

What?
$ quarto render 8584.qmd
ERROR: Validation of YAML front matter failed.
ERROR: In file 8584.qmd
(line 2, columns 8--12) Field "title" has value 2000, which must instead be a string
1: ---
2: title: 2000
         ~~~~~
3: format: html
✖ The value 2000 is of type number.
ℹ The error happened in location title.

ERROR: Render failed due to invalid YAML.
````

**Comment by bryanwhiting** [comment]
GitHub.com/bryanwhiting/blog.

Just replace the book-reviews/1984.md title with a number instead of a number in strings.

My CI/CD file has the quarto version number. Maybe it's an old issue, so maybe it should be ignored. But running "quarto render index.qmd" was failing with this issue oddly. 🤷🏼

- Bryan
________________________________

**Reply by cscheid** [reply]
You're running 1.4.515, a pretty recent version (although 1.4.549 is the stable release).

I think this is a bug where we're not validating the document when building ancillary information for a website (incidentally, this is why we ask for a full reproduction instead of a limited one, so that we're able to pinpoint exactly what's going on).

**Reply by cscheid** [reply]
See #8594. I've improved the error message to now say:

```
ERROR: about.qmd: Title must be a string, but was number
```

That's still not ideal because we lack line number information and so on, but it should be an improvement over the currently bad error message.

**Reply by bryanwhiting** [reply]
Interesting, I just read your commit! Funny how such small things can end up being so complicated. Thanks for taking a look at it!

**Comment by bryanwhiting** [comment]
Got this error again today even with the latest quarto

**Reply by bryanwhiting** [reply]
My 1984 file got messed up again. So making that a string solved it

**Reply by cscheid** [reply]
Again I can't reproduce (but maybe you meant that your file ended up with a title that was a number somehow?)

**Reply by bryanwhiting** [reply]
No worries, yes - a git merge between systems caused the old 1984 integer to return instead of "1984", but I used the latest quarto and didn't get your new error message. 

Probably just my own issue and not worth your valuable time - thanks for all your support!
